src/space/objects/realization/loop-hole.js

```diff
@@ -0,0 +1,124 @@
+import Vector from "../../utils/vector";
+import EncounterObject from "../pattern/encounterObject";
+import Type from "../pattern/encounterObjectType";
+import MyMath from "../../utils/myMath";
+import AssetLoader from "../../utils/assetLoader";
+
+const LH_SPRITE_RADIUS = 20;
+const LH_RENDER_RADIUS = 30;
+const LH_MIN_DIS = 700;
+//LH_ENTRY_SOURCE_Y is the y coordinate of position in loophole png file
+const LH_ENTRY_SOURCE_Y = 40;
+//LH_EXIT_SOURCE_Y is the y coordinate of position in loophole png file
+const LH_EXIT_SOURCE_Y = 120;
+// the change value of globalAlpha each time
+const LH_FADE_DEGREE = 5;
+
+
+export default class LoopHole extends EncounterObject{
+
+    /**
+      * Construction for a new loop hole
+      *
+      * @param {Encounter} encounter
+      * @param {Vector} position
+      */
+    constructor(encounter, position){
+        super(encounter, Type.LOOP_HOLE, LH_RENDER_RADIUS, position);
+	    this.mapWidth = encounter.width;
+	    this.mapHeight = encounter.height;
+        // generate loop hole exit
+      	do{
+      		this.outX = Math.floor(Math.random()*this.mapWidth);
+      		this.outY = Math.floor(Math.random()*this.mapHeight);
+            this.outPos = new Vector(this.outX, this.outY);
+      	} while (MyMath.distance(position, this.outPos) <= LH_MIN_DIS);
+        //load image
```

Comment:
Please inspect code of this method in Github UI ... Do you really think your indentation is ok? o.O

![image](https://user-images.githubusercontent.com/42660075/49833043-f7718d00-fd5d-11e8-82ac-0278b3256c22.png)